src/elements/content-explorer/ContentExplorer.js

```diff
@@ -383,6 +393,44 @@ class ContentExplorer extends Component<Props, State> {
         }
     };
 
+    /**
+     * Call Box Representations API to fetch thumbnails of all files in collection.
+     * items in currentCollection will have their thumbnailUrl field populated if a
+     * thumbnailUrl could be found.
+     *
+     * @private
+     * @param {Array<BoxItem>} items - item collection object
+     * @return {void}
+     */
+    fetchThumbnailUrls(items: Array<BoxItem> = []): void {
+        const fileAPI = this.api.getFileAPI();
+
+        items.forEach((item, index) => {
+            fileAPI.getFileThumbnail(item, index, this.getFileThumbnailSuccessCallback, this.errorCallback);
+        });
+    }
+
+    /**
+     * Callback called on success of getFileThumbnail. item at index in currentCollection
+     * will be populated with thumbnailUrl.
+     *
+     * @param {string} thumbnailUrl - url pointing towards thumbnail of item
+     * @param {number} index - index of item in currentCollection.items
+     * @return {void}
+     */
+    getFileThumbnailSuccessCallback = (thumbnailUrl: ?string, index: number) => {
+        const currentCollection = { ...this.state.currentCollection };
+        if (getProp(currentCollection, `items[${index}]`)) {
+            currentCollection.items = currentCollection.items.map((item, idx) => {
+                if (idx === index && thumbnailUrl) {
+                    return { ...item, thumbnailUrl };
```

Comment:
Realizing now that `item` will run into the same issue with modifying state.  This function will likely be removed soon though since the thumbnails will be fetched with the folder.